packages/core/src/api/guild.ts

```diff
@@ -0,0 +1,869 @@
+	 *
+	 * @param guildId - The id of the guild to fetch the roles fo
+	 */
+	public async getRoles(guildId: Snowflake) {
+		return this.rest.get(Routes.guildRoles(guildId)) as Promise<RESTGetAPIGuildRolesResult>;
+	}
+
+	/**
+	 * Creates a guild role
+	 *
+	 * @param guildId - The id of the guild to create the role in
+	 * @param data - The data to create the role with
+	 * @param reason - The reason for creating the role
+	 */
+	public async createRole(guildId: Snowflake, data: RESTPostAPIGuildRoleJSONBody, reason?: string) {
+		return this.rest.post(Routes.guildRoles(guildId), { reason, body: data }) as Promise<RESTPostAPIGuildRoleResult>;
+	}
+
+	/**
+	 * Sets a role position in a guild
+	 *
+	 * @param guildId - The id of the guild to set role positions for
+	 * @param data - The data for setting a role position
+	 * @param reason - The reason for setting the role position
+	 */
+	public async setRolePosition(guildId: Snowflake, data: RESTPatchAPIGuildRolePositionsJSONBody, reason?: string) {
+		return this.rest.patch(Routes.guildRoles(guildId), {
+			reason,
+			body: data,
+		}) as Promise<RESTPatchAPIGuildRolePositionsResult>;
+	}
+
+	/**
+	 * Edits a guild role
+	 *
+	 * @param guildId - The id of the guild to edit the role in
+	 * @param roleId - The id of the role to edit
+	 * @param data - data for editing the role
+	 * @param reason - The reason for editing the role
+	 */
+	public async editRole(guildId: Snowflake, roleId: string, data: RESTPatchAPIGuildRoleJSONBody, reason?: string) {
+		return this.rest.patch(Routes.guildRole(guildId, roleId), {
+			reason,
+			body: data,
+		}) as Promise<RESTPatchAPIGuildRoleResult>;
+	}
+
+	/**
+	 * Deletes a guild role
+	 *
+	 * @param guildId - The id of the guild to delete the role in
+	 * @param roleId - The id of the role to delete
+	 * @param reason - The reason for deleting the role
+	 */
+	public async deleteRole(guildId: Snowflake, roleId: string, reason?: string) {
+		await this.rest.delete(Routes.guildRole(guildId, roleId), { reason });
+	}
+
+	/**
+	 * Edits the multi-factor-authentication (MFA) level of a guild
+	 *
+	 * @param guildId - The id of the guild to edit the MFA level for
+	 * @param level - The new MFA level
+	 * @param reason - The reason for editing the MFA level
+	 */
+	public async editMFALevel(guildId: Snowflake, level: number, reason?: string) {
+		return this.rest.post(Routes.guild(guildId), {
+			reason,
+			body: { mfa_level: level },
+		}) as Promise<RESTPostAPIGuildsMFAResult>;
+	}
+
+	/**
+	 * Fetch the number of pruned members in a guild
+	 *
+	 * @param guildId - The id of the guild to fetch the number of pruned members from
+	 * @param options - The options for fetching the number of pruned members
+	 */
+	public async getPruneCount(guildId: Snowflake, options: { days?: number; includeRoles?: string[] } = {}) {
+		return this.rest.get(Routes.guildPrune(guildId), {
+			query: makeURLSearchParams(options as Record<string, unknown>),
```

Comment:
This won't pass includeRoles correctly as it expects snake case

Reply:
This should just use `RESTGetAPIGuildPruneCountQuery` instead